apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.component.ts

```diff
@@ -65,28 +66,38 @@ export class CreateOrUpdateAccessDialog implements OnDestroy {
     this.dialogRef.close();
   }
 
-  public onSubmit() {
+  public async onSubmit() {
     const access: CreateAccessDto = {
       alias: this.accessForm.controls['alias'].value,
       granteeUserId: this.accessForm.controls['userId'].value,
       permissions: [this.accessForm.controls['permissions'].value]
     };
 
-    this.dataService
-      .postAccess(access)
-      .pipe(
-        catchError((error) => {
-          if (error.status === StatusCodes.BAD_REQUEST) {
-            alert($localize`Oops! Could not grant access.`);
-          }
-
-          return EMPTY;
-        }),
-        takeUntil(this.unsubscribeSubject)
-      )
-      .subscribe(() => {
-        this.dialogRef.close({ access });
+    try {
+      await validateObjectForForm({
+        classDto: CreateAccessDto,
+        form: this.accessForm,
+        object: access
       });
+
+      this.dataService
+        .postAccess(access)
+        .pipe(
+          catchError((error) => {
+            if (error.status === StatusCodes.BAD_REQUEST) {
+              alert($localize`Oops! Could not grant access.`);
+            }
+
+            return EMPTY;
+          }),
+          takeUntil(this.unsubscribeSubject)
+        )
+        .subscribe(() => {
+          this.dialogRef.close({ access });
```

Comment:
This is currently not consistent:
`this.dialogRef.close({ access });`
vs.
`this.dialogRef.close(platform);`

What do you prefer?

Reply:
To be honest I kind of prefer the second approach so I've updated the access dialog close to match.